### PR TITLE
fix: allow version 9 of the babel-loader peer dependency

### LINKED ISSUE #24435: Upgrade webpack-preprocessor for babel-loader 9
### Current behavior

After the new release of `babel-loader` `9.0.0` the installation on a project with `webpack-preprocessor` produces a dependency error.

Screenshot 2022-10-28 at 12 14 16

https://github.com/cypress-io/cypress/blob/develop/npm/webpack-preprocessor/package.json#L40

### Desired behavior

The installation of the newer babel version should not cause package install issues.

### Test code to reproduce

Install `babel-loader` 9 in a project with the latest `@cypress/webpack-preprocessor`

### Cypress Version

10.11.0

### Node version

v19.0.0

### Operating System

mac Ventura 13

### Debug Logs

```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @cypress/webpack-preprocessor@5.15.0
npm ERR! Found: babel-loader@9.0.0
npm ERR! node_modules/babel-loader
npm ERR!   dev babel-loader@"9.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer babel-loader@"^8.0.2" from @cypress/webpack-preprocessor@5.15.0
npm ERR! node_modules/@cypress/webpack-preprocessor
npm ERR!   dev @cypress/webpack-preprocessor@"5.15.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: babel-loader@8.2.5
npm ERR! node_modules/babel-loader
npm ERR!   peer babel-loader@"^8.0.2" from @cypress/webpack-preprocessor@5.15.0
npm ERR!   node_modules/@cypress/webpack-preprocessor
npm ERR!     dev @cypress/webpack-preprocessor@"5.15.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```
```


### Other

_No response_

### PR DESCRIPTION
**Pull request by arnaugiralt**
- Closes [24435](https://github.com/cypress-io/cypress/issues/24435)

### Additional details
There is a dependency issue between babel-loader and @cypress/webpack-preprocessor. The installation of this newer babel-loader version should not cause package install issues. The user should be able to install both v8 and v9 of @babel/loader.

### Steps to test
Just forcing an install with babel-loader@^9 is enough to test that everything works.
Since the main breaking change in [v9](https://github.com/babel/babel-loader/releases/tag/v9.0.0) is dropping Webpack v4 and @cypress/webpack-preprocessor already allows for v5, I think it is safe to allow this change. If necessary, we could also update the @babel/core peer dependency to ^7.12, but babel-loader already has it declared as a peer dependency, so as you see fit.

### PR Tasks
- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by ZachJW34 — APPROVED** [review]
Change makes sense, we don't want to block users if they want to use newer deps.

LGTM!

**Review by mike-plummer — APPROVED** [review]
Thanks for the contribution!

**Comment by ZachJW34** [comment]
Holding off on merging as we have a planned patch release to fix a regression with 12.4.0. Once we get that out I'll merge this!

**Review by lmiller1990 — APPROVED** [review]

**Comment by arnaugiralt** [comment]
Hi team, thanks for the merge. Any ETA on a release for the @cypress/webpack-preprocessor npm package?

**Comment by ZachJW34** [comment]
@arnaugiralt It should be released with the next green develop build! (I'm assuming you're using `@cypress/webpack-preprocessor` in it's npm form which should be released automatically)

**Pull request merged**
